Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Cover/ObjectCoverPickerViewModel.swift

```diff
@@ -28,36 +27,67 @@ final class ObjectCoverPickerViewModel: ObservableObject {
     var isRemoveButtonAvailable: Bool { document.details?.documentCover != nil }
 
     // MARK: - Private variables
+    @Injected(\.objectHeaderUploadingService)
+    private var objectHeaderUploadingService: ObjectHeaderUploadingServiceProtocol
+    
     private let document: BaseDocumentGeneralProtocol
-    private let onCoverAction: (ObjectCoverPickerAction) -> Void
         
     // MARK: - Initializer
     
     init(data: ObjectCoverPickerData) {
         self.document = data.document
-        self.onCoverAction = data.onCoverAction
     }
 }
 
 extension ObjectCoverPickerViewModel {
     
     func setColor(_ colorName: String) {
-        onCoverAction(.setCover(.color(colorName: colorName)))
+        Task {
+            try await objectHeaderUploadingService.handleCoverAction(
```

Comment:
Now screen contains uploading logic. Not HeaderViewModel. Header just handle uploading state from ObjectHeaderUploadingServiceProtocol